tripleblindmarket/private-kit
# [SAF-272] - Safe Paths updates intersection logic to use hashes instead of GPS points

#### Description:

This PR changes the intersection logic used to determine if the app user has crossed paths with an infected person. It is an implementation of the algorithm described in [this Confluence doc](https://pathcheck.atlassian.net/wiki/spaces/TEST/pages/81231882/Hashing+of+GPS+points+-+Design) created by Diarmid Mackenzie.

This PR also includes logic for paginated fetching of HA data described in [SAF-275](https://pathcheck.atlassian.net/browse/SAF-275).



#### How to test:
- install the app and wait for it to record some local data points
- manually add a mock HA server url to the list of subscribed HAs
- wait for the app to trigger the intersection calculation
- if the random data published by the mock HA matches enough of your local data point hashes, you should get an exposure notification
